mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -2904,3 +2904,50 @@ fn test_parse_tx_with_huge_locktime() {
     let verbose_tx: RpcTransaction = json::from_str(verbose).expect("!json::from_str");
     let _: UtxoTx = deserialize(verbose_tx.hex.as_slice()).unwrap();
 }
+
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn test_withdraw_to_p2sh() {
+    UtxoStandardCoin::ordered_mature_unspents.mock_safe(|coin, _| {
+        let cache = block_on(coin.as_ref().recently_spent_outpoints.lock());
+        let unspents = vec![UnspentInfo {
+            outpoint: OutPoint {
+                hash: 1.into(),
+                index: 0,
+            },
+            value: 1000000000,
+            height: Default::default(),
+        }];
+        MockResult::Return(Box::pin(futures::future::ok((unspents, cache))))
+    });
+
+    let client = NativeClient(Arc::new(NativeClientImpl::default()));
+
+    let coin = utxo_coin_for_test(UtxoRpcClientEnum::Native(client.clone()), None);
+
+    // Create a p2sh address for the test coin
+    let checksum_type = ChecksumType::DSHA256;
+    let seed = "spice describe gravity federal blast come thank unfair canal monkey style afraid";
+    let key_pair = key_pair_from_seed(seed).unwrap();
+    let p2sh_address = Address {
+        prefix: 85,
+        hash: key_pair.public().address_hash(),
+        t_addr_prefix: 0,
+        checksum_type,
+        hrp: None,
+        addr_format: UtxoAddressFormat::Standard,
+    };
+
+    let withdraw_req = WithdrawRequest {
+        amount: 1.into(),
+        to: p2sh_address.to_string(),
+        coin: TEST_COIN_NAME.into(),
+        max: false,
+        fee: None,
+    };
+    let tx_details = coin.withdraw(withdraw_req).wait().unwrap();
+    let transaction: UtxoTx = deserialize(tx_details.tx_hex.as_slice()).unwrap();
+    let output_script: Script = transaction.outputs[0].script_pubkey.clone().into();
+
+    assert!(output_script.is_pay_to_script_hash());
```

Comment:
Could you please also add the test that withdrawal to P2PKH/P2WPKH addresses also generates a valid script pubkey? It's better to cover all the cases for this critical functionality. Also, please use `assert_eq!` to check that the script pubkey is exactly equal to the expected - it's possible to construct P2SH with the wrong hash and it will pass the test.